validator/client/src/test/java/tech/pegasys/teku/validator/client/restapi/apis/DeleteKeysTest.java

```diff
@@ -13,28 +13,94 @@
 
 package tech.pegasys.teku.validator.client.restapi.apis;
 
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
-import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_INTERNAL_SERVER_ERROR;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
+import static tech.pegasys.teku.validator.client.restapi.apis.schema.DeleteKeyResult.success;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.Collections;
 import java.util.List;
 import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import tech.pegasys.teku.bls.BLSPublicKey;
 import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
 import tech.pegasys.teku.spec.TestSpecFactory;
 import tech.pegasys.teku.spec.util.DataStructureUtil;
+import tech.pegasys.teku.validator.client.KeyManager;
+import tech.pegasys.teku.validator.client.restapi.apis.schema.DeleteKeysRequest;
+import tech.pegasys.teku.validator.client.restapi.apis.schema.DeleteKeysResponse;
 
 public class DeleteKeysTest {
-  private final DataStructureUtil dataStructureUtil =
+  final DataStructureUtil dataStructureUtil =
       new DataStructureUtil(TestSpecFactory.createMinimalAltair());
+  private final KeyManager keyManager = mock(KeyManager.class);
+  private final DeleteKeys endpoint = new DeleteKeys(keyManager);
+
+  private final DeleteKeysRequest requestData = new DeleteKeysRequest();
+  private final RestApiRequest request = mock(RestApiRequest.class);
+
+  @Test
+  @SuppressWarnings("unchecked")
+  void shouldCallKeyManagerWithListOfKeys() throws JsonProcessingException {
+    final ArgumentCaptor<List<BLSPublicKey>> captor = ArgumentCaptor.forClass(List.class);
+    final List<BLSPublicKey> keys =
+        List.of(dataStructureUtil.randomPublicKey(), dataStructureUtil.randomPublicKey());
+    requestData.setPublicKeys(keys);
+    when(keyManager.deleteValidators(eq(keys)))
+        .thenReturn(new DeleteKeysResponse(List.of(success(), success()), ""));
+    when(request.getRequestBody()).thenReturn(requestData);
+
+    endpoint.handle(request);
+    verify(keyManager).deleteValidators(captor.capture());
+
+    final List<BLSPublicKey> actual = captor.getValue();
+    assertThat(actual).isEqualTo(keys);
```

Comment:
Wouldn't it work to just say:
```
verify(keyManager).deleteValidators(keys);
```
Just makes things a lot simpler if we don't have to use a captor.

Reply:
hrm not sure, will give it a go, was making a mess when i tried...